spec/design.md

```diff
@@ -174,22 +180,32 @@ The return value is `constructor`, except it is an instance of class
 `Class`, which defines properties that describe the rest of the class.
 
 Example:
+
 ```{R}
-Child <- defineClass("Range", Vector, function(start, end) {
+Range <- defineClass("Range", 
+  Vector, 
+  function(start, end) {
     stopifnot(is.numeric(start), is.numeric(end), end >= start)
```

Comment:
Minor detail: to me, `end >= start` may blur the line between constructor and validator a bit.

Reply:
Some redundancy is good for defensive programming and more direct error messages.

Reply:
And I think the line between constructor and validator is fundamentally blurred 😄